### PR TITLE
llama 3.1 has correct `max_seq_len` for all versions

### PR DESCRIPTION
**Pull request by akashc1**
#### Context
What is the purpose of this PR? Is it to
- [ ] add a new feature
- [x] fix a bug
- [ ] update tests and/or documentation
- [ ] other (please add here)

Please link to any issues this PR addresses. #2202 

#### Changelog
What are the changes made in this PR?
* Ensure all llama 3.1 instantiations uses the correct `max_seq_len` the model was originally trained with based on the [HF config](https://huggingface.co/meta-llama/Llama-3.1-405B-Instruct/blob/main/config.json#L18)

#### Test plan
Please make sure to do each of the following if applicable to your PR. If you're unsure about any one of these just ask and we will happily help. We also have a [contributing page](https://github.com/pytorch/torchtune/blob/main/CONTRIBUTING.md) for some guidance on contributing.

- [x] run pre-commit hooks and linters (make sure you've first installed via `pre-commit install`)
- [ ] add [unit tests](https://github.com/pytorch/torchtune/tree/main/tests/torchtune) for any new functionality
- [ ] update [docstrings](https://github.com/pytorch/torchtune/tree/main/docs/source) for any new or updated methods or classes
- [ ] run unit tests via `pytest tests`
- [ ] run recipe tests via `pytest tests -m integration_test`
- [x] manually run any new or modified recipes with sufficient proof of correctness
- [ ] include relevant commands and any other artifacts in this summary (pastes of loss curves, eval results, etc.)

#### UX
If your function changed a public API, please add a dummy example of what the user experience will look like when calling it.
Here is a [docstring example](https://github.com/pytorch/torchtune/blob/6a7951f1cdd0b56a9746ef5935106989415f50e3/torchtune/modules/vision_transformer.py#L285)
and a [tutorial example](https://pytorch.org/torchtune/main/tutorials/qat_finetune.html#applying-qat-to-llama3-models)

- [x] I did not change any public API
- [ ] I have added an example to docs or docstrings

### CODE REVIEW

**Comment by felipemello1** [comment]
thanks for the PR, do you mind updating the docstrings too?!

**Comment by RdoubleA** [comment]
This seems inconsistent with our other builders, which do not specify a max seq len parameter. If you want to change the default max seq len for llama3_1 you should use the base builder `llama3_1()` instead of the 8b and 405b builders, as they're meant to expose minimal parameters and give all the defaults to get the correct model. But let us know if this is too prohibitive.

**Comment by akashc1** [comment]
@felipemello1 @RdoubleA thank you for the comments, can you please check the updated implementation? I set the correct default from the [HF model config](https://huggingface.co/meta-llama/Llama-3.1-405B-Instruct/blob/main/config.json#L18) for Llama 3.1 models in the `_model_builders` file.

**Comment by felipemello1** [comment]
I am on PTO this week, but if Rafi doesnt approve it by next week, i will review it. @RdoubleA , you said "If you want to change the default max seq len for llama3_1 you should use the base builder llama3_1()", that is true, but if the model is the LoRA, this parameter is not exposed. IMO, all model parameters should be exposed in the lora builder. Do you disagree?

**Comment by ebsmothers** [comment]
Commented on the issue as well, but I think we should only change the `llama3_1_405b` and `lora_llama3_1_405b` builders here. If we change `llama3_1` to hardcode `max_seq_len` it'd be inconsistent with our other models

**Comment by ebsmothers** [comment]
@akashc1 thanks for catching the bug. Do you want to push the changes I mentioned? (Basically your PR should just set `max_seq_len=131072` in `llama3_1_405b` and `lora_llama3_1_405b`.) If not let me know and I can push to your PR.

**Review by RdoubleA — APPROVED** [review]
